annotation/app/main.py

```diff
@@ -65,6 +68,17 @@ def get_version() -> str:
     dependencies=[Depends(TOKEN)],
 )
 
+logger = app_logger.Logger
+
+
+async def catch_exceptions_middleware(request: Request, call_next):
+    try:
+        return await call_next(request)
+    except Exception as exception:  # noqa
+        logger.exception(exception)
+        return Response("Internal server error", status_code=500)
```

Comment:
Shouldn't we re-raise an exception after logging? Because we may/will have a middleware to handle those exceptions on our own, at this point we simply conceal them.
```suggestion
    except Exception as exception:  # noqa
        logger.exception(exception)
        raise exception
```